### PR TITLE
Hw 24.1

### PR DESCRIPTION
**Pull request by MaliukDaria**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,61 @@
...
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 
 </project>
```
Comment (DmGrt) [inline pom.xml]: Add empty line)

File: src/main/java/com/online/store/service/ProductServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    @Inject
+    ProductDao productDao;
+
+    @Override
+    public Product create(Product product) {
+        productDao.create(product);
+        return product;
```
Comment (DmGrt) [inline src/main/java/com/online/store/service/ProductServiceImpl.java]: create() returns Product obj, so here you can write it in one line too

**Comment by DmGrt** [comment]
Good job!

File: src/main/java/com/online/store/service/ProductServiceImpl.java (hunk 2 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/service/ProductServiceImpl.java]: agree)

File: pom.xml (hunk 1 above)
Comment (MaliukDaria) [reply pom.xml]: done)

File: pom.xml [hunk 3]
```diff
@@ -8,5 +8,60 @@
...
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: let's remove this dependency - we don't use it

File: pom.xml [hunk 4]
```diff
@@ -8,5 +8,60 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 </project>
```
Comment (KseniiaMakarova) [inline pom.xml]: I can see the empty line is still needed

File: src/main/java/com/online/store/dao/OrderDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.online.store.dao;
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/dao/OrderDaoImpl.java]:
```suggestion
package com.online.store.dao.impl;
```
same for all dao implementations

File: src/main/java/com/online/store/dao/ProductDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    public Product create(Product product) {
+        Storage.addProduct(product);
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long productId) {
+        return Storage.products.stream()
+                .filter(p -> Objects.equals(p.getId(), (productId)))
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/dao/ProductDaoImpl.java]: let's not use Objects.equals()

File: src/main/java/com/online/store/dao/ProductDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                .findFirst();
+    }
+
+    @Override
+    public Product update(Product product) {
+        IntStream.range(0, Storage.products.size())
+                .filter(i -> Objects.equals(Storage.products.get(i).getId(), product.getId()))
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/dao/ProductDaoImpl.java]: same

File: src/main/java/com/online/store/dao/ProductDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+                .forEach(i -> Storage.products.set(i, product));
+        return product;
+    }
+
+    @Override
+    public boolean delete(Long productId) {
+        return Storage.products.removeIf(i -> Objects.equals(i.getId(), productId));
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/dao/ProductDaoImpl.java]: same

File: src/main/java/com/online/store/model/Order.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.util.List;
+
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public Order(List<Product> products, Long userId) {
+        this.userId = userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/model/Order.java:12]: let's initialize `products` here

File: src/main/java/com/online/store/model/ShoppingCart.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.util.List;
+
+public class ShoppingCart {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public ShoppingCart(List<Product> products, Long userId) {
+        this.userId = userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/model/ShoppingCart.java]: same

File: src/main/java/com/online/store/service/OrderServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,7 @@
+package com.online.store.service;
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/service/OrderServiceImpl.java]:
```suggestion
package com.online.store.service.impl;
```
same for all service implementations

File: src/main/java/com/online/store/service/OrderServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,7 @@
...
+
+import com.online.store.lib.Service;
+
+@Service
+public class OrderServiceImpl implements OrderService{
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/service/OrderServiceImpl.java]:
```suggestion
public class OrderServiceImpl implements OrderService {
```
same for all service implementations

File: src/main/java/com/online/store/service/ProductServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    ProductDao productDao;
+
+    @Override
+    public Product create(Product product) {
+        return productDao.create(product);
+    }
+
+    @Override
+    public Product get(Long productId) {
+        return productDao.get(productId).orElseThrow();
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/service/ProductServiceImpl.java]:
```suggestion
        return productDao.get(productId).get();
```
let's use `get()` for now

File: src/main/java/com/online/store/service/ProductServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+import com.online.store.dao.ProductDao;
+import com.online.store.lib.Inject;
+import com.online.store.lib.Service;
+import com.online.store.model.Product;
+import java.util.List;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    @Inject
+    ProductDao productDao;
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/service/ProductServiceImpl.java]: access modifier

File: src/main/java/com/online/store/service/ShoppingCardServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,7 @@
+package com.online.store.service;
+
+import com.online.store.lib.Service;
+
+@Service
+public class ShoppingCardServiceImpl implements ShoppingCartService{
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/service/ShoppingCardServiceImpl.java]:
```suggestion
public class ShoppingCartServiceImpl implements ShoppingCartService {
```

File: src/main/java/com/online/store/Application.java [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+        Product car = new Product("Audi", 100500);
+        Product notebook = new Product("Asus", 1500);
+        productService.create(phone);
+        productService.create(car);
+        productService.create(notebook);
+        System.out.println(productService.getAll());
+        productService.delete(car.getId());
+        System.out.println(productService.getAll());
+        Product anotherNotebook = new Product("Dell",300);
+        anotherNotebook.setId(3L);
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/store/Application.java]:
```suggestion
        anotherNotebook.setId(notebook.getId());
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (MaliukDaria) [reply pom.xml]: done

File: pom.xml (hunk 4 above)
Comment (MaliukDaria) [reply pom.xml]: done

File: src/main/java/com/online/store/dao/OrderDaoImpl.java (hunk 5 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/dao/OrderDaoImpl.java]: done

File: src/main/java/com/online/store/dao/ProductDaoImpl.java (hunk 6 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/dao/ProductDaoImpl.java]: ok, but why?

File: src/main/java/com/online/store/service/ProductServiceImpl.java (hunk 13 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/service/ProductServiceImpl.java]: done

File: src/main/java/com/online/store/service/ProductServiceImpl.java (hunk 14 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/service/ProductServiceImpl.java]: done

File: src/main/java/com/online/store/service/ShoppingCardServiceImpl.java (hunk 15 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/service/ShoppingCardServiceImpl.java]: done

File: src/main/java/com/online/store/Application.java (hunk 16 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/Application.java]: done

File: src/main/java/com/online/store/model/Order.java (hunk 9 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/model/Order.java:12]: done

File: src/main/java/com/online/store/model/ShoppingCart.java (hunk 10 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/model/ShoppingCart.java]: done

File: src/main/java/com/online/store/service/OrderServiceImpl.java (hunk 11 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/service/OrderServiceImpl.java]: done

File: src/main/java/com/online/store/service/OrderServiceImpl.java (hunk 12 above)
Comment (MaliukDaria) [reply src/main/java/com/online/store/service/OrderServiceImpl.java]: done

File: pom.xml [hunk 17]
```diff
@@ -8,5 +8,60 @@
...
-</project>
\ No newline at end of file
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (KseniiaMakarova) [inline pom.xml]: the dependency is still here

**Review by KseniiaMakarova — APPROVED** [review]

File: pom.xml (hunk 17 above)
Comment (MaliukDaria) [reply pom.xml]: sorry) Removed

**Pull request merged**